pkg/shell/topnav.jsx

```diff
@@ -64,6 +64,20 @@ export class TopNav extends React.Component {
         this.superuser = null;
 
         read_os_release().then(os => this.setState({ osRelease: os || {} }));
+
+        this.handleClickOutside = () => this.setState({ menuOpened: false, docsOpened: false });
+    }
+
+    componentDidMount() {
+        /* This is a HACK for catching lost clicks on the pages which live in iframes so as to close dropdown menus on the shell.
+         * Note: Clicks on an <iframe> element won't trigger document.documentElement listeners, because it's literally different page with different security domain.
+         * However, when clicking on an iframe moves focus to its content's window that triggers the main window.blur event.
+         */
+        window.addEventListener("blur", this.handleClickOutside);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener("blur", this.handleClickOutside);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17959-20221129-115044-2afa12d5-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)